aten/src/ATen/native/ComplexHelper.h

```diff
@@ -13,28 +13,43 @@ inline std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
   return res;
 }
 
+template<typename unused>
+inline void handle_unused() {}
+
+inline Tensor from_empty(
+    IntArrayRef sizes,
+    IntArrayRef strides,
+    const TensorOptions& options = {}) {
+  AutoNonVariableTypeMode guard;
+  auto storage = Storage(
+      options.dtype(),
+      detail::computeStorageSize(sizes, strides),
+      DataPtr(nullptr, options.device()),
+      /*allocator=*/nullptr,
+      /*resizable=*/false);
+  return at::empty({0}, options).set_(storage, 0, sizes, strides);
+}
+
 // expects as input a complex tensor and returns back a float tensor
 // containing the complex values in the last two dimensions
 inline Tensor view_complex_as_float(const Tensor& self) {
```

Comment:
this name is confusing -- it's not returning a "float" it's returning a "real" right?  I.e. the dimensions of the complex viewed as a their corresponding real components.